samples/callbacks_sample/src/main.c

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdio.h>
+
+#include "gateway_ll.h"
+
+void my_callback(GATEWAY_HANDLE gw, GATEWAY_EVENT event_type, GATEWAY_EVENT_CTX ctx)
+{
+    printf("Hello from callback!\n");
+}
+
+void second_callback(GATEWAY_HANDLE gw, GATEWAY_EVENT event_type, GATEWAY_EVENT_CTX ctx)
+{
+	printf("Hello from second callback!\n");
+}
+
+int main(int argc, char** argv)
+{
+    GATEWAY_HANDLE gw = Gateway_LL_Create(NULL);
+    Gateway_AddEventCallback(gw, GATEWAY_DESTROYED, my_callback);
+	Gateway_AddEventCallback(gw, GATEWAY_DESTROYED, second_callback);
+    printf("Hello with running gateway!\n");
+    Gateway_LL_Destroy(gw);
+	printf("Gateway was destroyed\n");
+}
```

Comment:
It's not possible to listen to more events right now other than the destroy one. There's a future TODO to seperate Start and Init of Gateway so we can register for creation. Additionally, more events will be added in the future. Will change printed messages though.